.travis.yml

```diff
@@ -32,15 +36,33 @@ matrix:
     env: TOXENV=py36
   - python: "3.7"
     env: TOXENV=py37
-  - python: "2.7"
-    env: TOXENV=py27-flake8
-  - python: "3.7"
-    env: TOXENV=py37-flake8
+  - os: osx
+    # No version of Python is available via virtualenv on OS X workers, see https://github.com/travis-ci/travis-ci/issues/2312
+    language: generic
+    env: TOXENV=py27
+  - os: osx
+    language: generic
+    env: TOXENV=py35
+  - os: osx
+    language: generic
+    env: TOXENV=py36
+  - os: osx
+    language: generic
+    env: TOXENV=py37
   - python: "pypy2.7-6.0"
     env: TOXENV=pypy
-
   fast_finish: true
 
+before_install:
+  - |
+    if [ "$TRAVIS_OS_NAME" == 'osx' -a "$TOXENV" != "py27" ]; then
+        curl -O https://repo.anaconda.com/miniconda/Miniconda2-latest-MacOSX-x86_64.sh
```

Comment:
Why conda?

Reply:
If you mean I could have used `pyenv`, I've tried it but unfortunately it is not possible to specify only Python major and minor versions (without patch), e.g. `pyenv install 3.5` does not work.

Reply:
So you use `miniconda` to install Python of the same version that we already have in the current tox environment?

Reply:
What would fail if we just drop `before_install`?

Reply:
Oh, I see — Travis doesn't fully support Python under MacOS. We have to install Python and create a virtualenv ourselves.

Reply:
Next question — why miniconda 2 when there is miniconda 3?

Reply:
Exactly, `language: python` is not available for `osx` builds. Miniconda 2 or 3 should not matter since we then create a conda environment with a specified Python version, but you are right that using Miniconda3 is more future-proof, I'll do that now.